openedx_learning/core/publishing/api.py

```diff
@@ -107,6 +162,37 @@ def learning_package_exists(key: str) -> bool:
     return LearningPackage.objects.filter(key=key).exists()
 
 
+def get_last_publish(learning_package_id: int) -> PublishLog | None:
+    return PublishLog.objects \
+                     .filter(learning_package_id=learning_package_id) \
+                     .order_by('-id') \
+                     .first()
+
+def get_all_drafts(learning_package_id: int):
+    return Draft.objects.filter(
+        entity__learning_package_id=learning_package_id,
+        version__isnull=False,
+    )
+
+def get_entities_with_unpublished_changes(learning_package_id: int):
+    return PublishableEntity.objects \
+               .filter(learning_package_id=learning_package_id) \
+               .exclude(draft__version=F('published__version'))
```

Comment:
Nit: super duper minor, but the 11-space indentation of these lines seems totally random. I guess it's four spaces from the start of `PublishableEntity` ? Which is inconsistent with `get_last_publish`.

Reply:
Yeah, I was trying out a few different things. I think establishing a decent convention is actually going to matter for this API because it has long function names and encourages chaining query sets.

How would you feel about this alternative?

```python
    return (
        PublishableEntity
            .objects
            .filter(
                learning_package_id=learning_package_id,
                draft__version__isnull=True,
            )
            .exclude(published__version__isnull=True)
    )
```

I think it looks a little funny at first, but it gives the most space for chaining purposes and long variable assignment, it doesn't require using backslashes.

Reply:
Turns out that this offends the linter. I guess I'll go with:

```python
def get_entities_with_unpublished_changes(learning_package_id: int):
    return PublishableEntity.objects \
                            .filter(learning_package_id=learning_package_id) \
                            .exclude(draft__version=F('published__version'))
```

Reply:
Darn, I did like that alternative. Too bad the linter didn't.